File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/bot/util/reply/Reply.java
package lunalauf.rms.utilities.network.bot.util.reply;

import org.telegram.telegrambots.meta.exceptions.TelegramApiException;

public abstract class Reply {
	
	protected final long chatId;
	protected final ReplyContainer wrapper;

	public Reply(long chatId, ReplyContainer wrapper) {
		this.chatId = chatId;
		this.wrapper = wrapper;
	}
	
	abstract public void runReplacedAction() throws TelegramApiException;

}

File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/bot/util/reply/Reply.kt
package lunalauf.rms.utilities.network.bot.util.reply

import org.telegram.telegrambots.meta.exceptions.TelegramApiException

abstract class Reply(
    protected val chatId: Long,
    protected val wrapper: ReplyContainer
) {
    @Throws(TelegramApiException::class)
    abstract fun runReplacedAction()
}
